demonstrations/tutorial_adaptive_circuits.py

```diff
@@ -0,0 +1,271 @@
+r"""
+
+Adaptive circuits for quantum chemistry
+=======================================
+
+.. meta::
+    :property="og:description": Learn how to build quantum chemistry circuits adaptively
+
+    :property="og:image": https://pennylane.ai/qml/_images/adaptive_circuits.png
+
+.. related::
+    tutorial_quantum_chemistry Quantum chemistry with PennyLane
+    tutorial_vqe A brief ofverview of VQE
+
+
+*Author: PennyLane dev team. Posted:  2021. Last updated: *
+
+Quantum computers promise a significant speedup in simulating molecular electronic structures for
+systems that are intractable with classical quantum chemistry algorithms. The variational quantum
+eigensolver (VQE) [#peruzzo2014]_, [#yudong2019]_ is the method of choice for performing such
+quantum chemistry simulations on near-term quantum devices. The key component of the VQE algorithm
+is to prepare the quantum state describing the ground electronic state of a molecule on a quantum
+device. Preparation of the quantum state requires a quantum circuit that is composed of a certain
+number of gates applied to an initial reference state. For a given molecule, the appropriate circuit
+can be generated by using a pre-selected wavefunction ansatz or it can be generated via a
+specialized approach that builds a customized ansatz for the given molecule. Using a pre-constructed
+ansatz has the disadvantage of reducing performance in favour of generality while building a
+customized circuit helps with improving performance in the cost of reducing generality.
+
+In this tutorial you will learn how to build customized quantum chemistry circuits adaptively. This
+includes a recipe for adaptive selection of the gates that have a significant contribution to the
+desired state and neglecting those which have zero or small contribution. You also learn to use
+the functionality in PennyLane for leveraging the sparsity of a molecular Hamiltonian to make the
+computation of the expectation values even more efficient. Let’s get started!
+
+Quantum chemistry circuits
+--------------------------
+
+Quantum chemistry circuits are typically build using a pre-generated wavefunction ansatz such as
+UCCSD. In this approach, one includes all possible single and double excitations of electrons from
+the occupied spin-orbitals of a reference state to the unoccupied spin-orbitals. This makes
+construction of the ansatz straightforward for any given molecule. However, we know that in
+practical applications, only a selected number of such excitations are necessary to prepare the
+exact ground state wavefunction and including all possible excitations increases the cost of the
+simulations without improving the accuracy of the results. This motivates implementing an strategy
+that allows approximating the contribution of the excitations and select only those excitations

```

Comment:
```suggestion
that allows approximating the contribution of the excitations and selecting only those excitations
```
What does "approximating the contribution of the excitations" means